src/exporter/DaemonMetricCollector.cc

```diff
@@ -121,42 +121,49 @@ void DaemonMetricCollector::dump_asok_metrics() {
     for (auto &perf_group_item : counter_schema) {
       std::string perf_group = {perf_group_item.key().begin(),
                                 perf_group_item.key().end()};
-      json_object perf_group_object = perf_group_item.value().as_object();
-      auto counters = perf_group_object["counters"].as_object();
-      auto counters_labels = perf_group_object["labels"].as_object();
-      auto counters_values =
-          counter_dump[perf_group].as_object()["counters"].as_object();
-      labels_t labels;
-
-      for(auto &label: counters_labels) {
-        std::string label_key = {label.key().begin(), label.key().end()};
-        labels[label_key] = quote(label.value().as_string().c_str());
-      }
-      for (auto &counter : counters) {
-        json_object counter_group = counter.value().as_object();
-        if (counter_group["priority"].as_int64() < prio_limit) {
-          continue;        
-        }
-        std::string counter_name_init =  {counter.key().begin(), counter.key().end()};
-        std::string counter_name = perf_group + "_" + counter_name_init;
-        promethize(counter_name);
-
-        if (counters_labels.empty()) {
-          auto labels_and_name = get_labels_and_metric_name(daemon_name, counter_name);
-          labels = labels_and_name.first;
-          counter_name = labels_and_name.second;
+      json_array perf_group_schema_array = perf_group_item.value().as_array();
+      json_array perf_grp_dump_array = counter_dump[perf_group].as_array();
+      for (auto schema_itr = perf_group_schema_array.begin(), dump_itr = perf_grp_dump_array.begin(); schema_itr != perf_group_schema_array.end() && dump_itr != perf_grp_dump_array.end(); ++schema_itr, ++dump_itr) {
+        auto& perf_grp = *schema_itr;
+        auto& perf_dump_grp = *dump_itr;
+        auto counters = perf_grp.at("counters").as_object();
+
+      for (auto &perf_grp : perf_group_schema_array) {
```

Comment:
> > This really looks like some kind of a leftover since you end up iterating twice...
> 
> @avanthakkar Since there are no tests, I'm asking you to post raw Prometheus endpoint data, showcasing metrics for at least two RBD images being mirrored, generated with this PR.

I'm pasting output for `counter dump` and metrics that are exported too
``` 
**>> counter dump:**
"rbd_mirror_snapshot_image": [
        {
            "labels": {
                "image": "image1",
                "namespace": "",
                "pool": "data"
            },
            "counters": {
                "snapshots": 1,
                "sync_time": {
                    "avgcount": 1,
                    "sum": 0.443453426,
                    "avgtime": 0.443453426
                },
                "sync_bytes": 52428800,
                "remote_timestamp": 1686770879.447388333,
                "local_timestamp": 1686770879.447388333,
                "last_sync_time": 0.443453426,
                "last_sync_bytes": 52428800
            }
        },
        {
            "labels": {
                "image": "image2",
                "namespace": "",
                "pool": "data"
            },
            "counters": {
                "snapshots": 1,
                "sync_time": {
                    "avgcount": 1,
                    "sum": 0.339616113,
                    "avgtime": 0.339616113
                },
                "sync_bytes": 52428800,
                "remote_timestamp": 1686772833.832611569,
                "local_timestamp": 1686772833.832611569,
                "last_sync_time": 0.339616113,
                "last_sync_bytes": 52428800
            }
        }
    ],
```
**>> `/metrics` endpoint:**
```ceph_rbd_mirror_snapshot_imagelast_sync_bytes{ceph_daemon="client.admin.289979",image="image1",namespace="",pool="data"} 52428800
ceph_rbd_mirror_snapshot_imagelast_sync_bytes{ceph_daemon="client.admin.289979",image="image2",namespace="",pool="data"} 52428800
# HELP ceph_rbd_mirror_snapshot_imagelast_sync_time Time taken to sync the last snapshot
# TYPE ceph_rbd_mirror_snapshot_imagelast_sync_time gauge
ceph_rbd_mirror_snapshot_imagelast_sync_time{ceph_daemon="client.admin.289979",image="image1",namespace="",pool="data"} 0.000000
ceph_rbd_mirror_snapshot_imagelast_sync_time{ceph_daemon="client.admin.289979",image="image2",namespace="",pool="data"} 0.000000
# HELP ceph_rbd_mirror_snapshot_imagelocal_timestamp Timestamp of the local snapshot
# TYPE ceph_rbd_mirror_snapshot_imagelocal_timestamp gauge
ceph_rbd_mirror_snapshot_imagelocal_timestamp{ceph_daemon="client.admin.289979",image="image1",namespace="",pool="data"} 1.686771
ceph_rbd_mirror_snapshot_imagelocal_timestamp{ceph_daemon="client.admin.289979",image="image2",namespace="",pool="data"} 1.686773
# HELP ceph_rbd_mirror_snapshot_imageremote_timestamp Timestamp of the remote snapshot
# TYPE ceph_rbd_mirror_snapshot_imageremote_timestamp gauge
ceph_rbd_mirror_snapshot_imageremote_timestamp{ceph_daemon="client.admin.289979",image="image1",namespace="",pool="data"} 1.686771
ceph_rbd_mirror_snapshot_imageremote_timestamp{ceph_daemon="client.admin.289979",image="image2",namespace="",pool="data"} 1.686773
# HELP ceph_rbd_mirror_snapshot_imagesnapshots Number of snapshots synced
# TYPE ceph_rbd_mirror_snapshot_imagesnapshots counter
ceph_rbd_mirror_snapshot_imagesnapshots{ceph_daemon="client.admin.289979",image="image1",namespace="",pool="data"} 1
ceph_rbd_mirror_snapshot_imagesnapshots{ceph_daemon="client.admin.289979",image="image2",namespace="",pool="data"} 1
```

Reply:
This doesn't look like a complete output to me, please paste the whole thing next time.  But it's enough to spot that there is an underscore missing in all of these: shouldn't `ceph_rbd_mirror_snapshot_imagelast_sync_bytes` be `ceph_rbd_mirror_snapshot_image_last_sync_bytes`?